File: conectaBrenannd-api/conectaBrenannd-api/src/main/java/com/conecta/brenannd/service/IIngressoService.java
package com.conecta.brenannd.service;

import com.conecta.brenannd.controller.dto.ValidacaoIngressoResponseDTO;
import com.conecta.brenannd.entity.Ingresso;

public interface IIngressoService {

	Ingresso save(final Ingresso entity);
	
	ValidacaoIngressoResponseDTO validarIngresso(String cpfToken);
	
	Ingresso update(final long id, final Ingresso entity);
	
	void delete(final long id);
}

File: conectaBrenannd-web/src/app/components/entity/ingresso/services/iingresso.service.ts
import { Observable } from 'rxjs';
import { Ingresso, ValidacaoIngressoResponse } from '../models/ingresso.models';

export interface IIngressoService {
  save(request: Ingresso): Observable<Ingresso>;

  update(id: number, request: Ingresso): Observable<Ingresso>;

  delete(id: number): Observable<void>;

  list(): Observable<Ingresso[]>;

  findByID(id: number): Observable<Ingresso>;

  findByCPF(cpf: string): Observable<Ingresso[]>;

  listByUser(): Observable<Ingresso[]>;

  findMyIngressoById(id: number): Observable<Ingresso>;

  validarIngresso(cpfToken: string): Observable<ValidacaoIngressoResponse>;
}
